app/uk/gov/hmrc/gform/sdes/datastore/DataStoreWorkItemService.scala

```diff
@@ -111,20 +116,43 @@ class DataStoreWorkItemService(
       FileAudit(correlationId)
     )
 
+  override def searchAll(
+    formTemplateId: Option[FormTemplateId],
+    status: Option[ProcessingStatus]
+  ): Future[SdesWorkItemPageData] =
+    doSearch(None, formTemplateId, status)
+
   override def search(
     page: Int,
     pageSize: Int,
     formTemplateId: Option[FormTemplateId],
     status: Option[ProcessingStatus]
+  ): Future[SdesWorkItemPageData] = {
+    val skip = page * pageSize
+    doSearch(Some((skip, pageSize)), formTemplateId, status)
+  }
+
+  private def doSearch(
+    maybeSkipAndPageSize: Option[(Int, Int)],
+    formTemplateId: Option[FormTemplateId],
+    status: Option[ProcessingStatus]
   ): Future[SdesWorkItemPageData] = {
     val queryByFormTemplateId = formTemplateId.fold(Filters.exists("_id"))(t => equal("item.formTemplateId", t.value))
     val query = status.fold(queryByFormTemplateId)(s => Filters.and(equal("status", s.name), queryByFormTemplateId))
     val sort = equal("receivedAt", -1)
 
-    val skip = page * pageSize
     for {
-      dmsWorkItem <-
-        dataStoreWorkItemRepo.collection.find(query).sort(sort).skip(skip).limit(pageSize).toFuture().map(_.toList)
+      dmsWorkItem <- maybeSkipAndPageSize match {
+                       case Some((skip, pageSize)) =>
+                         dataStoreWorkItemRepo.collection
+                           .find(query)
+                           .sort(sort)
+                           .skip(skip)
+                           .limit(pageSize)
+                           .toFuture()
+                           .map(_.toList)
+                       case None => dataStoreWorkItemRepo.collection.find(query).sort(sort).toFuture().map(_.toList)
+                     }
```

Comment:
Why is this change in this PR?

Reply:
I added searchAll interface the algebra and reused existing implementation for search and searchAll

Reply:
not needed. removed.